pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -759,12 +760,30 @@ func (c *Reconciler) createRun(ctx context.Context, rprt *resources.ResolvedPipe
 		},
 		Spec: v1alpha1.RunSpec{
 			Ref:                rprt.PipelineTask.TaskRef,
+			Spec:               rprt.PipelineTask.TaskSpec,
 			Params:             rprt.PipelineTask.Params,
 			ServiceAccountName: taskRunSpec.TaskServiceAccountName,
 			PodTemplate:        taskRunSpec.TaskPodTemplate,
 		},
 	}
-
+	if rprt.PipelineTask.TaskSpec != nil {
+		r.Spec.Spec = &v1beta1.EmbeddedTask{
+			TypeMeta: runtime.TypeMeta{
+				APIVersion: rprt.PipelineTask.TaskSpec.TypeMeta.APIVersion,
+				Kind:       rprt.PipelineTask.TaskSpec.TypeMeta.Kind,
+			},
+			Spec:     rprt.PipelineTask.TaskSpec.Spec,
+			Metadata: rprt.PipelineTask.TaskSpec.Metadata,
+			TaskSpec: rprt.PipelineTask.TaskSpec.TaskSpec,
+		}
+		j, err := json.Marshal(rprt.PipelineTask.TaskSpec.Spec)
+		if err != nil {
+			return nil, err
+		}
+		r.Spec.Spec.Spec = runtime.RawExtension{
+			Raw: j,
+		}
```

Comment:
I'm not sure I understand this. We set the `Run.Spec` in L763.
Why do we need to rebuild and `EmbeddedTask` from the `PipelineTask.TaskSpec` (which is an `EmbeddedTask` already)?

Reply:
My understanding is, we need to embed a `json.Marshal` of the `TaspSpec.Spec`. Please feel free to correct me.